### ISSUE TITLE
Puppet 4.x - check data types in selectors

### ISSUE THREAD

**Issue by ITler** [rodjek/puppet-lint#451]
Maybe, I missed something, but it seems puppetlint 1.1.0 doesn't support [selectors with data types](https://docs.puppetlabs.com/puppet/latest/reference/lang_data_type.html#cases) leading to *WARNING: unquoted string in selector*.

As a work-around, how could I disable this specific check with less infliction on other checks?

**Comment by rnelson0** [comment]
When testing with the snippet from the selector page, I receive different errors (removed unrelated issues):
```
class bar {
  $enable_real = $enable ? {
    Boolean => $enable,
    String  => str2bool($enable),
    Numeric => num2bool($enable),
    default => fail('Illegal value for $enable parameter'),
  }
}

$ be puppet-lint bar.pp
WARNING: top-scope variable being used without an explicit namespace on line 2
WARNING: top-scope variable being used without an explicit namespace on line 3
WARNING: top-scope variable being used without an explicit namespace on line 4
WARNING: top-scope variable being used without an explicit namespace on line 5
```
@ITler Can you replicate the unquoted string warning, or do you see this warning as well with version 2.0.0?

**Comment by rnelson0** [comment]
@daenney You tagged this as a Puppet 4 issue, can you replicate this with 2.0.1? I don't see the originally reported issue.

**Comment by daenney** [comment]
I didn't tag this as a Puppet 4 issue, you did: 
<img width="372" alt="screenshot 2016-08-19 00 11 40" src="https://cloud.githubusercontent.com/assets/569574/17792545/903734ba-65a1-11e6-8939-c734ae494262.png">. I linked it from the "Puppet 4 only RFC" because it's one of the issues that showed up at the time but that was on January the 7th of this year.

But yes, with puppet-lint 2.0.1 and the example you posted I only get the top scope variable issue which makes sense is `$enabled` isn't defined anywhere.
```
WARNING: top-scope variable being used without an explicit namespace on line 2
WARNING: top-scope variable being used without an explicit namespace on line 3
WARNING: top-scope variable being used without an explicit namespace on line 4
WARNING: top-scope variable being used without an explicit namespace on line 5
```

**Comment by rnelson0** [comment]
Ah I marked it Puppet 4 because of the reference from the VP Puppet 4 ticket, whoops :) I'm going to close this out then as it seems to be resolved.
